File: src/com/leetcode/Easy/IsomorphicStrings.java
package com.leetcode.Easy;

// 205. Isomorphic Strings
public class IsomorphicStrings {
    public boolean isIsomorphic(String s, String t) {
        int[] s1 = new int[256];
        int[] t1 = new int[256];
        for (int i = 0; i < s.length(); i++) {
            if (s1[s.charAt(i)] != t1[t.charAt(i)])
                return false;

            s1[s.charAt(i)] = i + 1;
            t1[t.charAt(i)] = i + 1;
        }
        return true;
    }
}

File: JavaScript/Easy/IsomorphicStrings.js
/**
 * 205. Isomorphic Strings
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
const isIsomorphic = function (s, t) {
  const s1 = new Array(256).fill(0);
  const t1 = new Array(256).fill(0);

  for (let i = 0; i < s.length; i++) {
    if (s1[s.charCodeAt(i)] !== t1[t.charCodeAt(i)]) {
      return false;
    }

    s1[s.charCodeAt(i)] = i + 1;
    t1[t.charCodeAt(i)] = i + 1;
  }

  return true;
};